site/src/pages/DeploySettingsPage/OAuth2AppsSettingsPage/EditOAuth2AppPageView.tsx

```diff
@@ -0,0 +1,300 @@
+import type * as TypesGen from "api/typesGenerated";
+import { Alert } from "components/Alert/Alert";
+import { ErrorAlert } from "components/Alert/ErrorAlert";
+import { CodeExample } from "components/CodeExample/CodeExample";
+import { CopyableValue } from "components/CopyableValue/CopyableValue";
+import { Header } from "components/DeploySettingsLayout/Header";
+import { ConfirmDialog } from "components/Dialogs/ConfirmDialog/ConfirmDialog";
+import { DeleteDialog } from "components/Dialogs/DeleteDialog/DeleteDialog";
+import { Loader } from "components/Loader/Loader";
+import { Stack } from "components/Stack/Stack";
+import { TableLoader } from "components/TableLoader/TableLoader";
+import { createDayString } from "utils/createDayString";
+import { OAuth2AppForm } from "./OAuth2AppForm";
+
+export type MutatingResource = {
+  updateApp: boolean;
+  createSecret: boolean;
+  deleteApp: boolean;
+  deleteSecret: boolean;
+};
+
+type EditOAuth2AppProps = {
+  app?: TypesGen.OAuth2ProviderApp;
+  isLoadingApp: boolean;
+  isLoadingSecrets: boolean;
+  // mutatingResource indicates which resources, if any, are currently being
+  // mutated.
+  mutatingResource: MutatingResource;
+  updateApp: (req: TypesGen.PutOAuth2ProviderAppRequest) => void;
+  deleteApp: (name: string) => void;
+  generateAppSecret: () => void;
+  deleteAppSecret: (id: string) => void;
+  secrets?: readonly TypesGen.OAuth2ProviderAppSecret[];
+  fullNewSecret?: TypesGen.OAuth2ProviderAppSecretFull;
+  ackFullNewSecret: () => void;
+  error?: unknown;
+};
+
+export const EditOAuth2AppPageView: FC<EditOAuth2AppProps> = ({
+  app,
+  isLoadingApp,
+  isLoadingSecrets,
+  mutatingResource,
+  updateApp,
+  deleteApp,
+  generateAppSecret,
+  deleteAppSecret,
+  secrets,
+  fullNewSecret,
+  ackFullNewSecret,
+  error,
+}) => {
+  const theme = useTheme();
+  const [searchParams] = useSearchParams();
+  const [showDelete, setShowDelete] = useState<boolean>(false);
+
+  return (
+    <>
+      <Stack
+        alignItems="baseline"
+        direction="row"
+        justifyContent="space-between"
+      >
+        <Header
+          title="Edit OAuth2 application"
+          description="Configure an application to use Coder as an OAuth2 provider."
+        />
+        <Button
+          component={Link}
+          startIcon={<KeyboardArrowLeft />}
+          to="/deployment/oauth2-provider/apps"
+        >
+          All OAuth2 Applications
+        </Button>
+      </Stack>
+
+      {fullNewSecret && (
+        <ConfirmDialog
+          hideCancel
+          open={Boolean(fullNewSecret)}
+          onConfirm={ackFullNewSecret}
+          onClose={ackFullNewSecret}
+          title="OAuth2 client secret"
+          confirmText="OK"
+          description={
+            <>
+              <p>
+                Your new client secret is displayed below. Make sure to copy it
+                now; you will not be able to see it again.
```

Comment:
Really like the messaging here; I think I'm going to copy it over to the Tokens tab in the user settings page